### PR TITLE
Add FolioClient as a dependency

### LINKED ISSUE #1: Add folio-client
Add the folio-client gem.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Add FolioClient gem as a dependency

Fixes #1

This commit adds the FolioClient as a dependency in all envs and adds the needed initializer.

Also includes commits for:
* Include `config` gem to hold Folio configuration
* Clean-up and organize Gemfile so it is easier to scan
* Fix GitHub PR template (wrong location)

### CODE REVIEW

**Review by justinlittman — COMMENTED** [review]
Why merge a commit that has its actions commented out? Would it make more sense to get the shared_configs set up and then merge?

**Comment by mjgiarlo** [comment]
@justinlittman 💬 
> Why merge a commit that has its actions commented out? Would it make more sense to get the shared_configs set up and then merge?

I understood #1 to include adding the gem to the Gemfile and that's all. So including the stub initializer is additive, which I included because we already know we're going to need it to use the client. I guess this PR could expand to include adding the `config` gem and the expected Folio config structure?

**Comment by justinlittman** [comment]
That makes sense to me @mjgiarlo.

**Comment by mjgiarlo** [comment]
@justinlittman 💬 
> That makes sense to me @mjgiarlo.

Done. Thx for the review!

File: config/initializers/folio_client.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+if Settings.folio.enabled
```
Comment (justinlittman) [inline config/initializers/folio_client.rb:3]: Do we need the enabled flag?

Comment (justinlittman) [reply config/initializers/folio_client.rb:3]: Oh, for test environment. Can you add a comment either here or in `settings.yml` to make that explicit?

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of request.

File: config/initializers/folio_client.rb (hunk 1 above)
Comment (mjgiarlo) [reply config/initializers/folio_client.rb]: Even better, I think? I removed the indirection. Have another look-see?

File: config/initializers/folio_client.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+if !Rails.env.test?
```
Comment (mjgiarlo) [inline config/initializers/folio_client.rb:3]: Took the config gem indirection out in favor of explicitness, @justinlittman

**Review by aaron-collier — APPROVED** [review]

**Pull request merged**